### PR TITLE
add kaldi to speech rec launch file

### PR DESCRIPTION
**Pull request by MatthijsBurgh**

### CODE REVIEW

**Comment by MatthijsBurgh** [comment]
@ar13pit ping ;)

File: launch/interaction/speech_recognition.launch [hunk 1]
```diff
@@ -4,6 +4,7 @@
 	<!-- Get the machine file -->
 	<arg name="machine" default="localhost"/>
         <arg name="dragonfly" default="true" />
+        <arg name="kaldi" default="false" />
```
Comment (ar13pit) [inline launch/interaction/speech_recognition.launch]: A couple of lines are missing:
```
    <!-- Kaldi model path default to ~/data/kaldi_speech_recognition -->
    <arg name="model_path" default="$(env HOME)/data/speech_models/model/"/>
    <param name="/kaldi_model_path" type="string" value="$(arg model_path)"/>
```

**Review by ar13pit — CHANGES_REQUESTED** [review]

File: launch/interaction/speech_recognition.launch [hunk 2]
```diff
@@ -3,7 +3,15 @@
...
 	<!-- Get the machine file -->
 	<arg name="machine" default="localhost"/>
-        <arg name="dragonfly" default="true" />
+    <arg name="dragonfly" default="true" />
+    <arg name="kaldi" default="false" />
+
+    <group if="$(arg kaldi)">
+        <!-- Kaldi model path default to ~/data/kaldi_speech_recognition -->
+        <arg name="model_path" default="$(env HOME)/data/speech_models/model/"/>
+        <param name="/kaldi_model_path" type="string" value="$(arg model_path)"/>
```
Comment (MatthijsBurgh) [inline launch/interaction/speech_recognition.launch]: @ar13pit why the absolute param path? This shouldn't be the case in my opinion

Comment (ar13pit) [reply launch/interaction/speech_recognition.launch]: In the `install.bash` file of the target, I download the trained model into this directory that's why I put it up this way.

If you have a better way of using this parameter then lets use that

Comment (MatthijsBurgh) [reply launch/interaction/speech_recognition.launch]: No it is about the param **name**.  This is not a param which is used by multiple nodes. So why is it a global param? I would like to see it as an private parameter.

Comment (ar13pit) [reply launch/interaction/speech_recognition.launch]: Definitely. It needs to be private and not global.

Comment (MatthijsBurgh) [reply launch/interaction/speech_recognition.launch]: @ar13pit Do you apply the change? This maybe also requires a change in the node

Comment (ar13pit) [reply launch/interaction/speech_recognition.launch]: You make the appropriate change here and I'll fix the node after that

Comment (MatthijsBurgh) [reply launch/interaction/speech_recognition.launch]: Done

**Review by ar13pit — APPROVED** [review]

**Pull request merged**